x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/jwt/JwtTestCase.java

```diff
@@ -363,48 +419,28 @@ public static OctetSequenceKey jwkHmacRemoveAttributes(final OctetSequenceKey hm
         return jwkGenerator;
     }
 
-    public static SignedJWT randomValidSignedJWT(final JWSSigner jwsSigner, final String signatureAlgorithm) throws Exception {
-        final String issuer = randomFrom("https://www.example.com/", "") + "iss1" + randomIntBetween(0, 99);
-        final List<String> audiences = randomFrom(List.of("rp_client1"), List.of("aud1", "aud2", "aud3"));
-        final String claimPrincipal = randomFrom("sub", "uid", "custom");
-        final String principal = "principal1";
-        final String claimGroups = randomBoolean() ? null : randomFrom("groups", "roles", "other");
-        final List<String> groups = randomFrom(List.of(""), List.of("grp1"), List.of("rol1", "rol2", "rol3"), List.of("per1"));
-        final SignedJWT unsignedJwt = randomValidJwsHeaderAndJwtClaimsSet(
-            signatureAlgorithm,
-            issuer,
-            audiences,
-            claimPrincipal,
-            principal,
-            claimGroups,
-            groups,
-            Map.of("metadata", randomAlphaOfLength(10))
-        );
-        return JwtValidateUtil.signJwt(jwsSigner, unsignedJwt);
-    }
-
-    public SecureString buildJWT(final JWSHeader header, final JWTClaimsSet claims, final Base64URL signature) throws ParseException {
-        final SignedJWT signedJwt = new SignedJWT(header.toBase64URL(), claims.toPayload().toBase64URL(), signature);
-        return new SecureString(signedJwt.serialize().toCharArray());
-    }
-
-    public static SignedJWT randomValidJwsHeaderAndJwtClaimsSet(
+    public static SignedJWT buildUnsignedJwt(
```

Comment:
Refactored to pass in all values, instead of generating some as random. This was required to generate a reproducible JWT for the realm config smoke test, with fixed claims like exp claim time 2099/1/1 12:00:00.000.